packages/web3-eth/test/integration/rpc.test.ts

```diff
@@ -0,0 +1,403 @@
+			expect(coinbase).toHaveLength(42);
+		});
+
+		it('isMining', async () => {
+			const isMining = await web3Eth.isMining();
+			expect(isMining).toBe(true);
+		});
+
+		it.each(Object.values(FMT_NUMBER))('getHashRate', async format => {
+			const hashRate = await web3Eth.getHashRate({
+				number: format as FMT_NUMBER,
+				bytes: FMT_BYTES.HEX,
+			});
+			expect(typeof hashRate).toBe(mapFormatToType[format as string]);
+		});
+
+		it('getAccounts', async () => {
+			const accList = await web3Eth.getAccounts();
+			expect(accList).toContain(accounts[0].toLowerCase());
+			expect(accList).toContain(accounts[1].toLowerCase());
+		});
+
+		it.each(Object.values(FMT_NUMBER))('getBlockNumber', async format => {
+			const res = await web3Eth.getBlockNumber({
+				number: format as FMT_NUMBER,
+				bytes: FMT_BYTES.HEX,
+			});
+			expect(typeof res).toBe(mapFormatToType[format as string]);
+			expect(parseInt(String(res), 16)).toBeGreaterThan(0);
+		});
+		it.each(Object.values(FMT_NUMBER))('getGasPrice', async format => {
+			const res = await web3Eth.getGasPrice({
+				number: format as FMT_NUMBER,
+				bytes: FMT_BYTES.HEX,
+			});
+			expect(typeof res).toBe(mapFormatToType[format as string]);
+			expect(parseInt(String(res), 16)).toBeGreaterThan(0);
+		});
+
+		it.each(Object.values(FMT_NUMBER))('getBalance', async format => {
+			const res = await web3Eth.getBalance(accounts[0], undefined, {
+				number: format as FMT_NUMBER,
+				bytes: FMT_BYTES.HEX,
+			});
+			expect(typeof res).toBe(mapFormatToType[format as string]);
+			expect(parseInt(String(res), 16)).toBeGreaterThan(0);
+		});
+
+		it('getStorageAt', async () => {
+			const numberData = 10;
+			const stringData = 'str';
+			const boolData = true;
+			await contract.methods?.setValues(numberData, stringData, boolData).send(sendOptions);
+			const resNumber = await web3Eth.getStorageAt(
+				contract.options.address as string,
+				0,
+				undefined,
+				{
+					number: FMT_NUMBER.BIGINT,
+					bytes: FMT_BYTES.HEX,
+				},
+			);
+			const resString = await web3Eth.getStorageAt(
+				contract.options.address as string,
+				1,
+				undefined,
+				{
+					number: FMT_NUMBER.STR,
+					bytes: FMT_BYTES.HEX,
+				},
+			);
+			const resBool = await web3Eth.getStorageAt(
+				contract.options.address as string,
+				2,
+				undefined,
+				{
+					number: FMT_NUMBER.NUMBER,
+					bytes: FMT_BYTES.HEX,
+				},
+			);
+
+			expect(hexToNumber(resNumber)).toBe(numberData);
+			// todo investigate why resString is not equal to stringData
+			expect(resString).toBeDefined();
+			expect(Boolean(hexToNumber(resBool))).toBe(boolData);
+		});
+
+		it.each(Object.values(FMT_NUMBER))('getCode', async format => {
+			const code = await web3Eth.getCode(contract?.options?.address as string, undefined, {
+				number: format as FMT_NUMBER,
+				bytes: FMT_BYTES.HEX,
+			});
+			expect(code).toBeDefined();
+			expect(BasicBytecode.slice(-100)).toBe(code.slice(-100));
+		});
+
+		it.each(
```

Comment:
```suggestion
// TODO: in next release add tests for getBlock with get block by hash and get block by number, also add test for validating all block fields (blocknum, hash, baseFeePerGas,nonce,stateRoot,sizegasLimit, ....etc) instead of parentHash only
		it.each(
```